assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,593 @@
+./R2:
+DPCh_plate1_A05_S5.R2.fq.gz   DPCh_plate1_C12_S36.R2.fq.gz  DPCh_plate1_F11_S71.R2.fq.gz
+DPCh_plate1_A06_S6.R2.fq.gz   DPCh_plate1_D05_S41.R2.fq.gz  DPCh_plate1_F12_S72.R2.fq.gz
+DPCh_plate1_A11_S11.R2.fq.gz  DPCh_plate1_D06_S42.R2.fq.gz  DPCh_plate1_G05_S77.R2.fq.gz
+DPCh_plate1_A12_S12.R2.fq.gz  DPCh_plate1_D11_S47.R2.fq.gz  DPCh_plate1_G06_S78.R2.fq.gz
+DPCh_plate1_B05_S17.R2.fq.gz  DPCh_plate1_D12_S48.R2.fq.gz  DPCh_plate1_G11_S83.R2.fq.gz
+DPCh_plate1_B06_S18.R2.fq.gz  DPCh_plate1_E05_S53.R2.fq.gz  DPCh_plate1_G12_S84.R2.fq.gz
+DPCh_plate1_B11_S23.R2.fq.gz  DPCh_plate1_E06_S54.R2.fq.gz  DPCh_plate1_H05_S89.R2.fq.gz
+DPCh_plate1_B12_S24.R2.fq.gz  DPCh_plate1_E11_S59.R2.fq.gz  DPCh_plate1_H06_S90.R2.fq.gz
+DPCh_plate1_C05_S29.R2.fq.gz  DPCh_plate1_E12_S60.R2.fq.gz  DPCh_plate1_H11_S95.R2.fq.gz
+DPCh_plate1_C06_S30.R2.fq.gz  DPCh_plate1_F05_S65.R2.fq.gz  DPCh_plate1_H12_S96.R2.fq.gz
+DPCh_plate1_C11_S35.R2.fq.gz  DPCh_plate1_F06_S66.R2.fq.gz
+
+#### Ex. 12: Use the recursive (-r) option to remove `lonely-fastqs`
+####         and its contents
+
+(base) [zhimei@colostate.edu@login11 lonely-fastqs]$ rm -r /projects/$USER/con-gen-csu/assignment
+s/002-unix-intro/lonely-fastqs/
+
+#### Ex. 13: Make two new directories in the top level of the
+####         repo called `fq_AB` and `fq_notAB`
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ mkdir fq_AB
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ mkdir fq_notAB
+
+#### Ex. 14: Using the `[` `]` copy the R1 files in `fastqs` that
+####         belong to sample names (the part immediately after `DPCh_plate1_`)
+####         starting with an `A` or `B` to `fq_AB`.
+
+(base) [zhimei@colostate.edu@login11 fastq]$ cp /projects/$USER/con-gen-csu/assignments/002-unix-intro/fastq/*DPCh_plate1_[A-B]* /projects/$USER/con-gen-csu/assignments/002-unix-intro/fq_AB
+
+(base) [zhimei@colostate.edu@login11 fastq]$ cd /projects/$USER/con-gen-csu/assignments/002-unix-intro/fq_AB
+(base) [zhimei@colostate.edu@login11 fq_AB]$ ls
+DPCh_plate1_A05_S5.R1.fq.gz   DPCh_plate1_B05_S17.R1.fq.gz
+DPCh_plate1_A05_S5.R2.fq.gz   DPCh_plate1_B05_S17.R2.fq.gz
+DPCh_plate1_A06_S6.R1.fq.gz   DPCh_plate1_B06_S18.R1.fq.gz
+DPCh_plate1_A06_S6.R2.fq.gz   DPCh_plate1_B06_S18.R2.fq.gz
+DPCh_plate1_A11_S11.R1.fq.gz  DPCh_plate1_B11_S23.R1.fq.gz
+DPCh_plate1_A11_S11.R2.fq.gz  DPCh_plate1_B11_S23.R2.fq.gz
+DPCh_plate1_A12_S12.R1.fq.gz  DPCh_plate1_B12_S24.R1.fq.gz
+DPCh_plate1_A12_S12.R2.fq.gz  DPCh_plate1_B12_S24.R2.fq.gz
+
+#### Ex. 15: Now, using negation (`^`) with `[` and `]` copy all the
+####         R1 files that don't belong to sample names starting with an
+####         `A` or a `B` to `fq_notAB`. When you are done, on the next line in
+####         the code block, list the contents of `fq_AB` and `fq_notAB`.
+
+(base) [zhimei@colostate.edu@login11 fq_notAB]$ cp /projects/$USER/con-gen-csu/assignments/002-unix-intro/fastq/*DPCh_plate1_[!A-B]* /projects/$USER/con
+-gen-csu/assignments/002-unix-intro/fq_notAB
+
```

Comment:
Check it out!  I have always used `^` for negation in these globs, but `!` works too!